purchases/src/main/kotlin/com/revenuecat/purchases/BillingWrapper.kt

```diff
@@ -319,17 +320,35 @@ internal class BillingWrapper internal constructor(
         }
     }
 
+    fun queryPurchaseHistoryBySku(
+        @SkuType skuType: String,
+        sku: String,
+        completion: (BillingResult, PurchaseHistoryRecordWrapper?) -> Unit
+    ) {
+        billingClient?.let { billingClient ->
```

Comment:
Note that if `billingClient` is null (that would be an edge case), this won't do anything. We do the same in other functions.

Reply:
I am worried that for example `startConnection` checks `if (billingClient == null)` and if it's null, it creates the `billingClient` object. So I am worried that doing something like this will confuse some people since it will display the warning everytime the field is accessed:

```
internal var billingClient: BillingClient? = null
        get() {
            if (field == null) debugLog("Warning: billing client is null, purchase methods won't work")
            return field
        }
```

I could make another property `getConnectedBillingClient` that's the one that displays the warning and only access the billingClient directly in `startConnection` and `endConnection`. Another option (which is the same actually) is to have another property `connectedBillingClient` which uses `billingClient` as a backing  field:

```
internal var connectedBillingClient: BillingClient? = null
        get() {
            if (billingClient == null) debugLog("Warning: billing client is null, purchase methods won't work")
            return billingClient
        }
```

But I think the winner option is:

```kotlin
   private fun withConnectedClient(receivingFunction: BillingClient.() -> Unit) {
        billingClient?.let {
            it.receivingFunction()
        } ?: debugLog("Warning: billing client is null, purchase methods won't work")
    }

    fun findPurchaseInPurchaseHistory(
        @SkuType skuType: String,
        sku: String,
        completion: (BillingResult, PurchaseHistoryRecordWrapper?) -> Unit
    ) {
        withConnectedClient { // this is billingClient
            debugLog("[QueryPurchases] Querying Purchase with $sku and type $skuType")
            queryPurchaseHistoryAsync(skuType) { result, purchasesList ->
                completion(
                    result,
                    purchasesList?.firstOrNull { sku == it.sku }?.let {
                        PurchaseHistoryRecordWrapper(it, PurchaseType.fromSKUType(skuType))
                    }
                )
            }
        }
    }
```

I think that's the cleanest

Reply:
Look at this https://github.com/RevenueCat/purchases-android/pull/166/commits/c7a2743d547f5f82332f8c36aa79ec12a22e76cc

Reply:
❤️  it